src/util/function.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * @flow
+ * @file Function utilities
+ * @author Box
+ */
+
+/**
+ * Wrapper around the promises.create() method to allow a promise to retry
+ * multiple times. A third parameter (besides resolve and reject) is passed
+ * that allows the function to cancel retrying and immediately reject.
+ * @param {function} func The function that performs the operation as a promise.
```

Comment:
The pattern that we are generally using for docs is like this

```js
/**
 * <Description>
 * <blank line>
 * @public / @private
 * @param {<type>} <name> - <description>
 * @return (singular) ...
 */
```

Reply:
done